test/e2e/windows/node_shutdown.go

```diff
@@ -0,0 +1,371 @@
+		if !kubelettypes.IsCriticalPod(pod) {
+			framework.Failf("pod %q should be a critical pod", pod.Name)
+		}
+	} else {
+		pod.Spec.PriorityClassName = priorityClassName
+		if kubelettypes.IsCriticalPod(pod) {
+			framework.Failf("pod %q should not be a critical pod", pod.Name)
+		}
+	}
+	return pod
+}
+
+// Emits a reboot event from HPC. Will cause kubelet to react to an active shutdown event.
+func emitSignalPrepareForShutdown(nodeName string, f *framework.Framework, ctx context.Context) {
+	ginkgo.By("scheduling a pod with a container that emits a PrepareForShutdown signal")
+
+	windowsImage := imageutils.GetE2EImage(imageutils.Agnhost)
+
+	trueVar := true
+	podName := "reboot-host-test-pod"
+	user := "NT AUTHORITY\\SYSTEM"
+	pod := &v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: podName,
+		},
+		Spec: v1.PodSpec{
+			SecurityContext: &v1.PodSecurityContext{
+				WindowsOptions: &v1.WindowsSecurityContextOptions{
+					HostProcess:   &trueVar,
+					RunAsUserName: &user,
+				},
+			},
+			HostNetwork: true,
+			Containers: []v1.Container{
+				{
+					Image: windowsImage,
+					Name:  "reboot-computer-test",
+					Command: []string{
+						"powershell.exe",
+						"-Command",
+						"$os = Get-WmiObject -Class win32_operatingsystem;",
+						"[Environment]::SetEnvironmentVariable(\"TMP_BOOT_DATE\", $os.LastBootUpTime, \"Machine\");",
+						"[Environment]::SetEnvironmentVariable(\"TMP_INSTALL_DATE\", $os.InstallDate, \"Machine\");",
+						"shutdown.exe -r -t 60",
+					},
+				},
+			},
+			RestartPolicy: v1.RestartPolicyNever,
+			NodeName:      nodeName,
+		},
+	}
+
+	e2epod.NewPodClient(f).Create(ctx, pod)
+
+	ginkgo.By("Waiting for pod to run")
+	e2epod.NewPodClient(f).WaitForFinish(ctx, podName, 3*time.Minute)
+
+	ginkgo.By("Then ensuring pod finished running successfully")
+	p, err := f.ClientSet.CoreV1().Pods(f.Namespace.Name).Get(
+		ctx,
+		podName,
+		metav1.GetOptions{})
+
+	framework.ExpectNoError(err, "Error retrieving pod")
+	gomega.Expect(p.Status.Phase).To(gomega.Equal(v1.PodSucceeded))
+}
+
+const (
+	// https://github.com/kubernetes/kubernetes/blob/1dd781ddcad454cc381806fbc6bd5eba8fa368d7/pkg/kubelet/nodeshutdown/nodeshutdown_manager_linux.go#L43-L44
+	podShutdownReason  = "Terminated"
+	podShutdownMessage = "Pod was terminated in response to imminent node shutdown."
+)
+
+func isPodShutdown(pod *v1.Pod) bool {
+	if pod == nil {
+		return false
+	}
+
+	hasContainersNotReadyCondition := false
+	for _, cond := range pod.Status.Conditions {
+		if cond.Type == v1.ContainersReady && cond.Status == v1.ConditionFalse {
+			hasContainersNotReadyCondition = true
+		}
+	}
+
+	return pod.Status.Message == podShutdownMessage && pod.Status.Reason == podShutdownReason && hasContainersNotReadyCondition && pod.Status.Phase == v1.PodFailed
+}
+
+// Pods should never report failed phase and have ready condition = true (https://github.com/kubernetes/kubernetes/issues/108594)
+func isPodStatusAffectedByIssue108594(pod *v1.Pod) bool {
```

Comment:
the issue is fixed. Why code needs to reference it?

Reply:
This is still in the linux related test: https://github.com/kubernetes/kubernetes/blob/88a997ac01cd8a386df9318d81b89257307f53fb/test/e2e_node/node_shutdown_linux_test.go#L254. I can remove it if this is no longer an issue